models/sipnet/R/model2netcdf.SIPNET.R

```diff
@@ -59,22 +59,27 @@ model2netcdf.SIPNET <- function(outdir, sitelat, sitelon, start_date, end_date,
     output[[5]] <- (sub.sipnet.output$rtot * 0.001) / timestep.s    # Total Respiration in kgC/m2/s
     output[[6]] <- (sub.sipnet.output$rAboveground * 0.001) / timestep.s +
       (sub.sipnet.output$rRoot * 0.001) / timestep.s                # Autotrophic Respiration in kgC/m2/s
-    output[[7]] <- (sub.sipnet.output$rSoil * 0.001) / timestep.s   # Heterotropic Respiration in kgC/m2/s
-    output[[8]] <- (sub.sipnet.output$nee * 0.001) / timestep.s     # NEE in kgC/m2/s
+    output[[7]] <- ((sub.sipnet.output$rSoil - sub.sipnet.output$rRoot) * 0.001) / timestep.s   # Heterotrophic Respiration in kgC/m2/s
+    output[[8]] <- (sub.sipnet.output$rSoil * 0.001) / timestep.s   # Soil Respiration in kgC/m2/s
+    output[[9]] <- (sub.sipnet.output$nee * 0.001) / timestep.s     # NEE in kgC/m2/s
     #output[[9]] <- rep(-999,sipnet.output.dims[1])             # CarbPools
-    output[[9]] <- (sub.sipnet.output$plantWoodC * 0.001)         # Above ground wood kgC/m2
-    output[[10]] <- (sub.sipnet.output$plantLeafC * 0.001)        # Leaf C kgC/m2
-    output[[11]] <- (sub.sipnet.output$plantWoodC * 0.001)+
+    output[[10]] <- (sub.sipnet.output$plantWoodC * 0.001)         # Above ground wood kgC/m2
+    output[[11]] <- (sub.sipnet.output$plantLeafC * 0.001)        # Leaf C kgC/m2
+    output[[12]] <- (sub.sipnet.output$plantWoodC * 0.001)+
       (sub.sipnet.output$plantLeafC * 0.001)+
       (sub.sipnet.output$coarseRootC * 0.001)+
       (sub.sipnet.output$fineRootC * 0.001)                       # Total living C kgC/m2
-    output[[12]] <- (sub.sipnet.output$soil * 0.001)+
+    output[[13]] <- (sub.sipnet.output$soil * 0.001)+
       (sub.sipnet.output$litter * 0.001)                          # Total soil C kgC/m2
-    output[[13]] <- (sub.sipnet.output$fluxestranspiration * 0.001) / timestep.s  #Transpiration kgW/m2/s
-    output[[14]] <- (sub.sipnet.output$soilWater * 10)            # Soil moisture kgW/m2
-    output[[15]] <- (sub.sipnet.output$soilWetnessFrac)         # Fractional soil wetness
-    output[[16]] <- (sub.sipnet.output$snow * 10)                 # SWE
-    output[[17]] <- sub.sipnet.output$litter * 0.001 ## litter kgC/m2
+    ## *** NOTE : npp in the sipnet output file is actually evapotranspiration, this is due to a bug in sipnet.c : ***
+    ## *** it says "npp" in the header (written by L774) but the values being written are trackers.evapotranspiration (L806) ***
+    ##  water density = 1000 kg m-3 , latent heat of vaporization = 2.501*10^6 J kg-1
+    output[[14]] <- (sub.sipnet.output$npp * 0.01 * 1000 * 2.501 * 10^6) / timestep.s  # Qle W/m2
```

Comment:
For clarity  I would leave the math explicit. I would hope R is smart enough to precompute constant math, but even still, the performance hit would be tiny.

If it helps a cm of water = cm^3 of water per cm^2 of area 

Latent heat of vaporization does vary slightly with temperature.  Here’s the equation we use in our flux processing code, where airtemp is degrees K, and Lv is in J/kg

 lv = (94.210*(365 - (airtemp-273.15))^0.31249)*4.183*1000.

For a constant, use 2.5e6

If you want to be even more accurate, note that density of water varies with temperature and pressure. For rho at 1 bar (10^5 Pa), could use Tanaka equation <https://answers.yahoo.com/question/index?qid=20120512001022AAvdzhC> 


> On Jun 2, 2016, at 11:43 AM, istfer <notifications@github.com> wrote:
> 
> In models/sipnet/R/model2netcdf.SIPNET.R <https://github.com/PecanProject/pecan/pull/858#discussion_r65575510>:
> 
> >        (sub.sipnet.output$litter * 0.001)                          # Total soil C kgC/m2
> > -    output[[13]] <- (sub.sipnet.output$fluxestranspiration * 0.001) / timestep.s  #Transpiration kgW/m2/s
> > -    output[[14]] <- (sub.sipnet.output$soilWater * 10)            # Soil moisture kgW/m2
> > -    output[[15]] <- (sub.sipnet.output$soilWetnessFrac)         # Fractional soil wetness
> > -    output[[16]] <- (sub.sipnet.output$snow * 10)                 # SWE
> > -    output[[17]] <- sub.sipnet.output$litter * 0.001 ## litter kgC/m2
> > +    ## *** NOTE : npp in the sipnet output file is actually evapotranspiration, this is due to a bug in sipnet.c : ***
> > +    ## *** it says "npp" in the header (written by L774) but the values being written are trackers.evapotranspiration (L806) ***
> > +    ##  water density = 1000 kg m-3 , latent heat of vaporization = 2.501*10^6 J kg-1
> > +    output[[14]] <- (sub.sipnet.output$npp * 0.01 * 1000 * 2.501 * 10^6) / timestep.s  # Qle W/m2
> I wrote this explicitly but should I reduce it to 2.501 * 10^7? Also I've seen different values for latent heat of vaporization (2.25x10^6, 2.3x10^6, 2.501x10^6 ...), which one should I use?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/PecanProject/pecan/pull/858/files/c818431257d081db0575998e895f3f77bd9e570b#r65575510>, or mute the thread <https://github.com/notifications/unsubscribe/AC8PlyZ8sUJ5IrnOdMLvWURb04DrAhUbks5qHwhAgaJpZM4IsukB>.
>